crypto/secp256k1/secp256.go

```diff
@@ -119,6 +121,33 @@ func RecoverPubkey(msg []byte, sig []byte) ([]byte, error) {
 	return pubkey, nil
 }
 
+// VerifySignature checks that the given pubkey created signature over message.
+// The signature should be in [R || S] format.
+func VerifySignature(pubkey, msg, signature []byte) bool {
+	if len(msg) != 32 || len(signature) != 64 || len(pubkey) == 0 {
```

Comment:
I'd prefer to leave it to libsecp256k1 to check for the length. If you disagree strongly we can add the check here too. The check for zero is needed to prevent a crash for the `&pubkey[0]` construction.

Reply:
The check [is done here](https://github.com/ethereum/go-ethereum/blob/master/crypto/secp256k1/libsecp256k1/src/eckey_impl.h#L17)